test/tensor.cpp

```diff
@@ -101,12 +101,67 @@ void test_read_write(const vector<T>& x)
     ASSERT_EQ(result1, result2);
 }
 
+template <typename T>
+void test_copy_to_same_backend(const vector<T>& x)
+{
+    auto backend_int = runtime::Backend::create("INTERPRETER");
+    auto a = backend_int->create_tensor(element::from<T>(), Shape{(x.size() / 2), (x.size() / 2)});
+    auto b = backend_int->create_tensor(element::from<T>(), Shape{(x.size() / 2), (x.size() / 2)});
+
+    vector<T> result(x.size());
+
+    a->write(&x[0], 0, x.size() * sizeof(T));
+    a->read(result.data(), 0, result.size() * sizeof(T));
+
+    auto s = a->get_size_in_bytes();
+    a->copy_to(b, 0, s);
+
+    vector<T> af_vector(x.size());
+    b->read(af_vector.data(), 0, af_vector.size() * sizeof(T));
+    ASSERT_EQ(af_vector, result);
+}
+
+template <typename T>
+void test_copy_to_other_backend(const vector<T>& x)
```

Comment:
Since there is nothing in copy_to related to backend then there is no need to test with two different backends. You only need to test that you can copy from one tensor to another.
I see no test for copy_to where the tensor sizes do not match. You can't just overwrite memory so you need to check for this condition in copy_to and you need a unit test to verify that you throw an exception when you try to copy to the wrong size tensor. You may want to compare size and shape.